wiring/inc/spark_wiring_asset.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2023 Particle Industries, Inc.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "hal_platform.h"
+#include "asset_manager_api.h"
+#include "spark_wiring_stream.h"
+#include <memory>
+#include "spark_wiring_vector.h"
+#include <functional>
+
+namespace particle {
+
+class ApplicationAsset: public Stream {
+public:
+    ApplicationAsset() = default;
+    ApplicationAsset(const asset_manager_asset* asset);
+    virtual ~ApplicationAsset() = default;
+
+    String name() const;
+    AssetHash hash() const;
+    size_t size() const;
+
+    bool isValid() const;
+    bool isReadable();
+
+    int available() override;
+    int read() override;
+    int peek() override;
+    // Not present in Stream
+    virtual int read(char* buffer, size_t size);
```

Comment:
For convenience, can we add an `uint8_t`  overload `virtual int read(uint8_t* buffer, size_t size);`?